### PR TITLE
feat: support adaLN layer while converting LoRAs from diffusers to flux

### PR DESCRIPTION
**Pull request by simpletrontdip**
## Summary

This PR updates loRA Diffusers -> Flux conversion logic based on its original source:
- Fix the key typos for `guidance_in` layer keys, as in: https://github.com/huggingface/diffusers/blob/55ac421f7bb12fd00ccbef727be4dc2f3f920abb/scripts/convert_flux_to_diffusers.py#L103-L115
- Add AdaLN layer from `norm_out` with **shift scale swapping**, as in : https://github.com/huggingface/diffusers/blob/55ac421f7bb12fd00ccbef727be4dc2f3f920abb/scripts/convert_flux_to_diffusers.py#L263-L268

## Related Issues / Discussions
I couldn't load `Hyper-FLUX.1-dev-Nsteps-lora.safetensors` from <https://huggingface.co/ByteDance/Hyper-SD> via InvokeUI.

## QA Instructions

1. Install [Hyper-FLUX 8 steps](https://huggingface.co/ByteDance/Hyper-SD/tree/main?show_file_info=Hyper-FLUX.1-dev-8steps-lora.safetensors), type `ByteDance/Hyper-SD` in the search box.
2. Load and run it with with Flux.dev flow, (to see it fail)
3. Apply the changes
4. Check the output

Screenshot 2025-02-28 at 15 01 56

## Merge Plan
Apply the change only, it's a small one

## Checklist

- [ ] _The PR has a short but descriptive title, suitable for a changelog_
- [ ] _Tests added / updated (if applicable)_
- [ ] _Documentation added / updated (if applicable)_
- [ ] _Updated `What's New` copy (if doing a release after this PR)_

### CODE REVIEW

File: invokeai/backend/patches/lora_conversions/flux_diffusers_lora_conversion_utils.py [hunk 1]
```diff
@@ -82,6 +82,19 @@ def add_lora_layer_if_present(src_key: str, dst_key: str) -> None:
...
+    def add_adaLN_lora_layer_if_present(src_key: str, dst_key: str) -> None:
+        if src_key in grouped_state_dict:
+            src_layer_dict = grouped_state_dict.pop(src_key)
+            values = get_lora_layer_values(src_layer_dict)
+            
+            for _key in values.keys():
+                # in SD3 original implementation of AdaLayerNormContinuous, it split linear projection output into shift, scale;
+                # while in diffusers it split into scale, shift. Here we swap the linear projection weights in order to be able to use diffusers implementation
+                scale, shift = values[_key].chunk(2, dim=0)
+                values[_key] = torch.cat([shift, scale], dim=0)
```
Comment (RyanJDick) [inline invokeai/backend/patches/lora_conversions/flux_diffusers_lora_conversion_utils.py:94]: This doesn't look right to me. If I'm understanding correctly, in the case of a vanilla LoRA layer, we should only be flipping one of the LoRA components.

The required transformation would be a bit more involved for other LoRA variants (LoHA, LoKR, etc.), so I'm fine with only supporting vanilla LoRAs. But, we should assert that the result of `any_lora_layer_from_state_dict()` is a `LoRALayer`.

**Review by RyanJDick — COMMENTED** [review]
Thanks for digging into this!

To get it merged, we'll need to:
1. Fix the shift/scale transformation.
2. Add a unit test in `test_flux_diffusers_lora_conversion_utils.py` for this new LoRA format. See the other tests in that file for reference.

File: invokeai/backend/patches/lora_conversions/flux_diffusers_lora_conversion_utils.py (hunk 1 above)
Comment (simpletrontdip) [reply invokeai/backend/patches/lora_conversions/flux_diffusers_lora_conversion_utils.py:94]: @RyanJDick Thanks for spending time on this. 
(Tbh, it turns out harder than I expected, sorry for that 👯 ) 

Back to the topic, as in vanilla LoRA format, we have A, and B that `d_W = w_A x w_B`, so there will be no `shift/scale` normalization involved as in normal SD model, so I guest no flipping needed.

If that's correct, my plan is to check the vanilla loRA format and throw if otherwise, create the corresponding layer with no flipping. And add tests to verify. 

Will that work?

Comment (simpletrontdip) [reply invokeai/backend/patches/lora_conversions/flux_diffusers_lora_conversion_utils.py:94]: Hi @RyanJDick thanks for spending time 👯 
I have to confess, it is more complex than I expected, sorry for not asking the team before hand.

As my understanding
```py
# for normal LoRA layer

delta_W = up @ down
W = W + delta_W

# for AdaLN in diffusers
W_prime = swap_shift_scale(W)
delta_W_prime = swap_shift_scale(delta_W)

# => We may need to add a custom LoRA layer to swap them in `get_weight`

class AdaLN_LoRALayer(LoRALayer):
   def get_weight(self, orig_weight: torch.Tensor) -> torch.Tensor:
       '''swap shift and scale before returning real weight'''
        weight = super().get_weight(orig_weight)
        scale, shift = weight.chunk(2, dim=0)
        return torch.cat([shift, scale], dim=0)

# we need to build and return this layer in our function
```

What do you think?

Comment (simpletrontdip) [reply invokeai/backend/patches/lora_conversions/flux_diffusers_lora_conversion_utils.py]: To fully make LoRA layer compatible with `CustomLinear` I've change the approach to approximate that final layer with SVD decomposition, added the tests.
But still feel a little hacky on this :D

**Comment by JPPhoto** [comment]
@simpletrontdip If you're still around, can you update and let us know if this is an issue? If so, would you be willing to take another look to make sure this PR is still relevant?

File: invokeai/backend/patches/lora_conversions/flux_diffusers_lora_conversion_utils.py (hunk 1 above)
Comment (JPPhoto) [reply invokeai/backend/patches/lora_conversions/flux_diffusers_lora_conversion_utils.py]: Are you still around? Is this still an issue in the latest `main`? Please provide guidance.

**Comment by JPPhoto** [comment]
@simpletrontdip Please respond to the prior question. If you don't, we'll have to close this PR up as stale.

**Comment by JPPhoto** [comment]
Closing due to inactivity.
